misc/python/materialize/checks/actions.py

```diff
@@ -58,8 +59,8 @@ def execute(self, e: Executor) -> None:
 
 
 class Initialize(Action):
-    def __init__(self, checks: List[Type["Check"]]) -> None:
-        self.checks = [check_class() for check_class in checks]
+    def __init__(self, checks: List[Type["Check"]], base_version: MzVersion) -> None:
```

Comment:
I think I designed the class hierarchy a bit incorrectly when this was started :-(

But I think this would be better:

```
class Initialize(Action):
    def __init__(self, scenario: Scenario) -> None:
        self.checks = [check_class(scenario.base_version()) for check_class in scenario.checks()]

class Scenario:
     def base_version(self) -> MzVersion:
          return self.cargo_version()
class UpgradeScenario(Scenario):
     def base_version(self) -> MzVersion:
          return ...

class UpgradeEntireMz(UpgradeScenario):
    ....
```

This way all mentions of `Initialize`'s constructor and the constructors of all other similar classes will have just a single argument going forward. `StartMz()`